src/object/player.cpp

```diff
@@ -444,6 +506,149 @@ Player::update(float dt_sec)
 
 }
 
+void
+Player::handle_input_swimming()
+{
+  float pointx = 0, pointy = 0;
+
+  if (m_controller->hold(Control::UP)&&!m_controller->hold(Control::DOWN))
+    pointy = -1.f;
+  else if (m_controller->hold(Control::DOWN)&&!m_controller->hold(Control::UP))
+    pointy = 1.f;
+  if (m_controller->hold(Control::LEFT)&&!m_controller->hold(Control::RIGHT))
+    pointx = -1.f;
+  else if (m_controller->hold(Control::RIGHT)&&!m_controller->hold(Control::LEFT))
+    pointx = 1.f;
+  bool boost = m_controller->pressed(Control::JUMP);
+
+  swim(pointx,pointy,boost);
+}
+
+void
+Player::swim(float pointx, float pointy, bool boost)
```

Comment:
Please add some documentation what all these calculations actually do. I guess they could be structured more clearly (the mix of changing vx vy and setting them more than once is a little bit confusing.) Idea: Compute the new velocities and set them just once.

Also I guess a lot of these constants should be declared as constants somewhere.